compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/text/input/internal/TextInputSession.skiko.kt

```diff
@@ -112,13 +131,152 @@ internal actual suspend fun PlatformTextInputSession.platformSpecificTextInputSe
 private fun TextFieldCharSequence.toTextFieldValue() =
     TextFieldValue(toString(), selection, composition)
 
+private fun TextFieldCharSequence.asTextEditorState() = object : TextEditorState {
+
+    override val length: Int
+        get() = this@asTextEditorState.length
```

Comment:
why we need to override this?

Reply:
This whole wrapper is because `TextFieldCharSequence` is in foundation, but `PlatformTextInputService(2)` is in ui. So foundation needs to wrap it in an interface that's available in ui.